service/frontend/service.go

```diff
@@ -94,6 +94,8 @@ type Config struct {
 	WorkerBuildIdSizeLimit                                            dynamicconfig.IntPropertyFn
 	ReachabilityTaskQueueScanLimit                                    dynamicconfig.IntPropertyFn
 	ReachabilityQueryBuildIdLimit                                     dynamicconfig.IntPropertyFn
+	ReachabilityCacheOpenWFsTTL                                       dynamicconfig.DurationPropertyFn
```

Comment:
Let's add namespace filter just in case we need it to override for a single namespace

Reply:
A namespace filter would require refactoring how the cache is instantiated when matching engine is instantiated, because we don't have knowledge of the namespace at that point. We do know the namespace when we query for reachability, so one way I could imagine supporting namespace filters is to have `matching_enging.reachabilityCache == nil` at first, and then do `if e.reachabilityCache == nil { e.reachabilityCache = newCache(ns) }` right before the call to `getReachability`.

Do you see another better way to handle it / do you think that workaround is worth it for the flexibility?

Reply:
sidenote: I'm pretty sure that there is only one namespace per matching engine (but possibly multiple matching engines per namespace if it's a multi-tenant namespace)

Reply:
Matching engine can have partitions belonging to many namespaces. It seems the only way to support ns level ttl would be having a map of ns->cache in matching engine. So let's forget about it.